### PR TITLE
Plans: Add Jetpack Anti-Spam to the Plans page

### PR DESCRIPTION
**Pull request by rcanepa**
#### Changes proposed in this Pull Request:

* Add Jetpack Anti-Spam product as a new Jetpack product so users can purchase it from the Plans page.

#### Jetpack product discussion

* Ticket: 1181531115069428-as-1182646983386225

#### Does this pull request change what data or activity we track or use?

* No. It doesn't change what we currently track or use.

#### Testing instructions:

* Run this PR locally or use Jetpack Beta Tester plugin.
* Select a site that doesn't have Jetpack Anti-Spam.
* Go to Jetpack Dashboard -> Plans.
* Verify that you see Jetpack Anti-Spam card.
* Verify that you can purchase Jetpack Anti-Spam monthly and Jetpack Anti-Spam yearly subscriptions.

image

#### Proposed changelog entry for your changes:
* Add Jetpack Anti-Spam to the Plans page so users can purchase it.

### CODE REVIEW

File: class.jetpack.php [hunk 1]
```diff
@@ -7451,6 +7451,24 @@ public static function get_products_for_purchase() {
...
+			'options'           => array(
+				array(
+					'type'      => 'anti-spam',
+					'slug'      => 'jetpack-anti-spam',
+					'key'       => 'jetpack_anti_spam',
+					'name'      => __( 'Anti-Spam', 'jetpack' ),
+				),
+			),
+			'default_option'    => 'anti-spam',
+			'included_in_plans' => array( 'anti-spam-plan' ),
```
Comment (rcanepa) [inline class.jetpack.php:7469]: Is this correct? Is Jetpack Anti-Spam included in any Jetpack plans?

Comment (ChaosExAnima) [reply class.jetpack.php:7469]: Looking at the code and testing, it seems this needs to include every paid plan class: `personal-plan`, `premium-plan`, etc.

**Review by ChaosExAnima — CHANGES_REQUESTED** [review]
Seeing it able to be purchased still with a Personal plan active. This should be prevented.

**Comment by rcanepa** [comment]
> Seeing it able to be purchased still with a Personal plan active. This should be prevented.

This should be fixed now.

**Review by ChaosExAnima — APPROVED** [review]
Tested and works great!

**Review by jeherve — APPROVED** [review]

**Pull request merged**